tests/garage/tf/models/test_model.py

```diff
@@ -0,0 +1,178 @@
+import pickle
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.mlp import mlp
+from garage.tf.models.base import TfModel
+from tests.fixtures import TfGraphTestCase
+
+
+class SimpleModel(TfModel):
+    def __init__(self, output_dim=2, hidden_sizes=(4, 4), name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+
+    def network_output_spec(self):
+        return ['state', 'action']
+
+    def _build(self, obs_input):
+        state = mlp(obs_input, self._output_dim, self._hidden_sizes, 'state')
+        action = mlp(obs_input, self._output_dim, self._hidden_sizes, 'action')
+        return state, action
+
+
+class ComplicatedModel(TfModel):
+    def __init__(self, output_dim=2, name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._simple_model_1 = SimpleModel(output_dim=4)
+        self._simple_model_2 = SimpleModel(
+            output_dim=output_dim, name='simple_model_2')
+
+    def network_output_spec(self):
+        return ['action']
+
+    def _build(self, obs_input):
+        h1, _ = self._simple_model_1.build(obs_input)
+        return self._simple_model_2.build(h1)[1]
```

Comment:
Could you show me the computation graph of this ComplicatedModel?

Reply:
<img width="810" alt="screen shot 2019-02-18 at 9 24 33 pm" src="https://user-images.githubusercontent.com/10270515/52991950-aae0a880-33c3-11e9-88cf-cf5df6bcabc4.png">